### PR TITLE
Security

### PR DESCRIPTION
**Pull request by KitNaIn**
basic security with roles

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                                response.sendError(
+                                        HttpStatus.UNAUTHORIZED.value(),
+                                        HttpStatus.UNAUTHORIZED.getReasonPhrase()
+                                )))
+                .authorizeHttpRequests(auth -> {
+                    auth.requestMatchers(
+                            ("/api/user/**")).permitAll();
+                    auth.requestMatchers(
+                            ("/api/**")).authenticated();
+                    auth.anyRequest().permitAll();
```
Comment (MJaschewski) [inline backend/src/main/java/de/neuefische/backend/security/SecurityConfig.java:43]: change to authenticated

File: backend/src/main/java/de/neuefische/backend/security/UserService.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        CrewUser newUser = CrewUser.builder()
+                .id(generateUUIDService.generateUUID())
+                .username(user.getUsername())
+                .password(passwordEncoder.encode(user.getPassword()))
+                .build();
+
+        if (userRepo.findUserByUsername(newUser.getUsername()).isPresent()) {
+            throw new IllegalArgumentException("User with username " + user.getUsername() + " already exists");
+        }
+        CrewUser temp = userRepo.save(newUser.withRoles(Collections.singletonList(new SimpleGrantedAuthority("user"))).withPassword(passwordEncoder.encode(user.getPassword())));
```
Comment (MJaschewski) [inline backend/src/main/java/de/neuefische/backend/security/UserService.java:44]: dont use temp. use a better name

**Review by MJaschewski — CHANGES_REQUESTED** [review]

File: backend/src/main/java/de/neuefische/backend/security/SecurityConfig.java (hunk 1 above)
Comment (KitNaIn) [reply backend/src/main/java/de/neuefische/backend/security/SecurityConfig.java]: changed

File: backend/src/main/java/de/neuefische/backend/security/UserService.java (hunk 2 above)
Comment (KitNaIn) [reply backend/src/main/java/de/neuefische/backend/security/UserService.java]: replaced

**Review by MJaschewski — APPROVED** [review]

**Pull request merged**